ruma-events-macros/src/event.rs

```diff
@@ -371,6 +375,37 @@ fn expand_from_into(
     }
 }
 
+fn expand_eq_ord_event(input: &DeriveInput, fields: &[Field]) -> Option<TokenStream> {
+    if fields.iter().flat_map(|f| f.ident.as_ref()).any(|f| f == "event_id") {
+        let ident = &input.ident;
+        let (impl_gen, ty_gen, where_clause) = input.generics.split_for_impl();
+
+        Some(quote! {
+            impl #impl_gen ::std::cmp::PartialEq for #ident #ty_gen #where_clause {
+                fn eq(&self, other: &Self) -> bool {
+                    self.event_id == other.event_id
+                }
+            }
+
+            impl #impl_gen ::std::cmp::Eq for #ident #ty_gen #where_clause {}
+
+            impl #impl_gen ::std::cmp::PartialOrd for #ident #ty_gen #where_clause {
+                fn partial_cmp(&self, other: &Self) -> ::std::option::Option<::std::cmp::Ordering> {
+                    self.event_id.partial_cmp(&other.event_id)
+                }
+            }
+
+            impl #impl_gen ::std::cmp::Ord for #ident #ty_gen #where_clause {
+                fn cmp(&self, other: &Self) -> ::std::cmp::Ordering {
+                    self.event_id.cmp(&other.event_id)
+                }
+            }
+        })
+    } else {
+        None
```

Comment:
Also not unambiguous.

Reply:
This is not part of the `quote!` block so it's ok right?

Reply:
Oh, of course 😄